src/libraries/Microsoft.Extensions.DependencyInjection.Specification.Tests/src/ActivatorUtilitiesTests.cs

```diff
@@ -350,6 +378,39 @@ public void GetServiceOrCreateInstanceUnregisteredService()
             }
         }
 
+        [Fact]
+        public void CreateInstance_BadlyConfiguredIServiceProviderIsService_ProperlyCreatesUnambiguousInstance()
```

Comment:
I think this test is more appropriate in the `DI.Tests\ActivatorUtilitiesTests.cs` file. This is more about testing ActivatorUtilities, and not the DI Implementations.

Also - what happens when all the external providers SupportsIServiceProviderIsService=true? Now this test no longer does anything.

Instead, I think we should make a mock `IServiceProvider` that returns an `IServiceProviderIsService` that always returns `false`. And then use that to test the fallback in ActivatorUtilities.